### PR TITLE
Re-suppress tolerable manual compaction stress test failures

### PR DESCRIPTION
**Pull request by hx235**
**Context/Summary:**
Previously manual compaction stress test failures won't terminate stress test. https://github.com/facebook/rocksdb/pull/12414 made more manual compaction failures terminate the stress test for signal boosting. A downside to that PR: some tolerable manual compaction stress test failures also unnecessarily terminate stress test. 

Ideally we should exclude exactly those tolerable errors (left as a TODO) from being able to terminate. For now we approximate those errors by Aborted(), InvalidArgument(), NotSupported() etc. It's still an improvement to pre-https://github.com/facebook/rocksdb/pull/12414 situation.

**Test:**
- No more tolerable manual compaction stress test failures terminating stress test.

### CODE REVIEW

File: db_stress_tool/db_stress_test_base.cc [hunk 1]
```diff
@@ -2528,19 +2530,17 @@ void StressTest::TestCompactRange(ThreadState* thread, int64_t rand_key,
...
+                               std::strstr(status.getState(), "injected") ||
+                               status.IsInvalidArgument() ||
+                               status.IsNotSupported();
+  if (!status.ok() && !non_ok_status_allowed) {
     fprintf(stdout,
             "Unable to perform CompactRange(): %s under specified "
             "CompactRangeOptions: %s (Empty string or "
             "missing field indicates default option or value is used)\n",
             status.ToString().c_str(), compact_range_opt_oss.str().c_str());
-    thread->stats.AddErrors(1);
```
Comment (hx235) [inline db_stress_tool/db_stress_test_base.cc:2543]: Mistakenly added in https://github.com/facebook/rocksdb/pull/12414 previously

File: db_stress_tool/db_stress_test_base.cc [hunk 2]
```diff
@@ -2528,19 +2530,17 @@ void StressTest::TestCompactRange(ThreadState* thread, int64_t rand_key,
...
     // Fail fast to preserve the DB state.
     thread->shared->SetVerificationFailure();
   }
```
Comment (cbi42) [inline db_stress_tool/db_stress_test_base.cc:2546]: Print those allowed non-ok status to stdout?

Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc:2546]: Sure

File: db_stress_tool/db_stress_test_base.cc [hunk 3]
```diff
@@ -2327,11 +2327,13 @@ void StressTest::TestCompactFiles(ThreadState* thread,
...
                                  static_cast<int>(output_level));
-      if (!s.ok() && !s.IsManualCompactionPaused() && !s.IsAborted() &&
-          !std::strstr(s.getState(), "injected") &&
-          !(s.IsInvalidArgument() &&
-            std::strstr(s.getState(), "Specified compaction input file") &&
-            std::strstr(s.getState(), "does not exist in column family"))) {
+      // TOOD (hx235): allow an exact list of tolerable failures under stress
+      // test
+      bool non_ok_status_allowed = s.IsManualCompactionPaused() ||
+                                   std::strstr(s.getState(), "injected") ||
```
Comment (cbi42) [inline db_stress_tool/db_stress_test_base.cc:2333]: it seems state can be nullptr when s is ok

Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc]: Fixed

File: db_stress_tool/db_stress_test_base.cc (hunk 2 above)
Comment (hx235) [reply db_stress_tool/db_stress_test_base.cc:2550]: Fixed

**Review by cbi42 — APPROVED** [review]

**Pull request closed without merging**